moon build --verbose fails on macOS arm64: missing _main when linking internal/raylib

### Summary
`moon build --verbose` fails in this repo on macOS arm64 with missing `_main` while linking `internal/raylib/raylib.exe`.

### Reproduced commit(s)
- `origin/main` (`2ae71de`)
- `fix/warnings` (`f9886ec`)

### Environment
- `moon 0.1.20260224 (2288f02 2026-02-24)`
- Feature flags: `rr_moon_pkg`
- macOS Darwin 25.0.0 (arm64)
- Apple clang 17.0.0

### Steps to reproduce
```bash
git clone https://github.com/moonbit-community/tonyfettes-raylib.git
cd tonyfettes-raylib
git checkout 2ae71de
moon version
moon build --verbose
```

### Actual behavior
`moon build --verbose` fails at link step:
```text
failed: /usr/bin/cc -o .../_build/native/debug/build/internal/raylib/raylib.exe ...
Undefined symbols for architecture arm64:
  "_main", referenced from:
      <initial-undefines>
ld: symbol(s) not found for architecture arm64
```

`moon build --dry-run --verbose` includes:
```text
moonc link-core ... -main tonyfettes/raylib/internal/raylib -o .../internal/raylib/raylib.c
```

### Expected behavior
`moon build` should complete for this repo, or docs should clearly specify the supported build command for contributors (if `moon build` is intentionally not supported for this module layout).

### Notes
- `moon test --target native` passes on the same machine (`Total tests: 81, passed: 81, failed: 0`).
- Full logs + environment snapshot: https://gist.github.com/bobzhang/82a0a6902f3be02488afd42d96d53d8e
- Originally reported in moonbitlang/moon#1488; moved here for repo-level tracking.

--- a/CLAUDE.md
+++ b/CLAUDE.md
@@ -8,6 +8,10 @@ MoonBit bindings for raylib 5.5 (`tonyfettes/raylib`). Native-only target (no WA
 
 ## Build Commands
 
+> **Note:** `moon build` (without arguments) will fail with a linker error — this
+> module is a library, not an executable. Always build a specific example package
+> or use `moon check --target native` to type-check.
+
 ```bash
 # Build an example (native only) — examples/ is a separate module
 moon -C examples build --target native raylib_demo/

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -1,0 +1,34 @@
+# Contributing
+
+## Prerequisites
+
+- [MoonBit toolchain](https://www.moonbitlang.com/download/) installed
+- Native target support (macOS, Linux, or Windows)
+
+## Building
+
+This module is a **library**, not an executable. Running `moon build` without
+arguments will fail with a linker error (missing `_main`). Use one of the
+following commands instead:
+
+```bash
+# Type-check without building
+moon check --target native
+
+# Build a specific example
+moon -C examples build --target native raylib_demo/
+
+# Run a specific example
+moon -C examples run --target native raylib_tank_1990/
+
+# Run tests
+moon test --target native
+```
+
+The `examples/` directory is a separate module, so `-C examples` is required to
+set the working directory.
+
+## Conventions
+
+- Use conventional commits (e.g., `feat:`, `fix:`, `refactor:`)
+- See [CLAUDE.md](CLAUDE.md) for architecture details, FFI rules, and coding patterns

